crates/unleash-edge-metrics/src/send_unleash_metrics.rs

```diff
@@ -62,6 +64,125 @@ fn get_valid_token(token_cache: Arc<TokenCache>) -> Option<EdgeToken> {
         .map(|t| t.clone())
 }
 
+#[derive(Debug)]
+enum MetricsSendError {
+    NoBackoff(String),
+    Backoff(String),
+    Unauthed(String),
+    Unknown(String),
+}
+
+async fn send_metrics(
```

Comment:
Same treatment as above - we split the error handling from the backoff calculations so that we can reuse this as a background task. This has has the consequence that the background task will not correctly batch the metrics if they're too large rather than yeeting them on the floor if the final batch is too large